flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/source/enumerator/BigQuerySourceEnumStateSerializerTest.java

```diff
@@ -0,0 +1,96 @@
+ * the License.
+ */
+
+package com.google.cloud.flink.bigquery.source.enumerator;
+
+import com.google.cloud.flink.bigquery.source.split.BigQuerySourceSplit;
+import com.google.cloud.flink.bigquery.source.split.BigQuerySourceSplitSerializer;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+/** */
+public class BigQuerySourceEnumStateSerializerTest {
+
+    private BigQuerySourceEnumState create() {
+
+        List<String> remainingTableStreams = new ArrayList<>();
+
+        remainingTableStreams.add("third stream");
+        remainingTableStreams.add("fourth stream");
+        remainingTableStreams.add("fifth stream");
+
+        List<String> completedTableStreams = new ArrayList<>();
+        completedTableStreams.add("first stream");
+
+        List<BigQuerySourceSplit> remainingSourceSplits = new ArrayList<>();
+        remainingSourceSplits.add(new BigQuerySourceSplit("second stream", 0L));
+
+        Map<String, BigQuerySourceSplit> assignedSourceSplits = new TreeMap<>();
+        assignedSourceSplits.put("key1", remainingSourceSplits.get(0));
+
+        return new BigQuerySourceEnumState(
+                remainingTableStreams,
+                completedTableStreams,
+                remainingSourceSplits,
+                assignedSourceSplits,
+                true);
+    }
+
+    @Test
+    public void testEnumStateSerializerInitialState() throws IOException {
+        BigQuerySourceEnumState initialState = BigQuerySourceEnumState.initialState();
+
+        byte[] serialized = BigQuerySourceEnumStateSerializer.INSTANCE.serialize(initialState);
+
+        BigQuerySourceEnumState enumState1 =
+                BigQuerySourceEnumStateSerializer.INSTANCE.deserialize(
+                        BigQuerySourceSplitSerializer.VERSION, serialized);
+
+        Assert.assertEquals(initialState, enumState1);
+        Assert.assertEquals(initialState.hashCode(), enumState1.hashCode());
+    }
+
+    @Test
+    public void testEnumStateSerializer() throws IOException {
+        BigQuerySourceEnumState enumState = create();
+
+        byte[] serialized = BigQuerySourceEnumStateSerializer.INSTANCE.serialize(enumState);
+
+        BigQuerySourceEnumState enumState1 =
+                BigQuerySourceEnumStateSerializer.INSTANCE.deserialize(
+                        BigQuerySourceSplitSerializer.VERSION, serialized);
+
+        Assert.assertEquals(enumState, enumState1);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testWrongSerializerVersion() throws IOException {
+        BigQuerySourceEnumState enumState = create();
+
+        byte[] serialized = BigQuerySourceEnumStateSerializer.INSTANCE.serialize(enumState);
+
+        BigQuerySourceEnumStateSerializer.INSTANCE.deserialize(1000, serialized);
+
+        // should never reach here
+        assert (true);
```

Comment:
Should we fail here if we don't expect the code reach here?

Reply:
lol, that was the intent, I will correct this.